src/main/java/org/jenkinsci/test/acceptance/po/Job.java

```diff
@@ -188,10 +209,18 @@ public void copyFile(File file) {
             ZipUtil.pack(file, tmp);
             byte[] archive = IOUtils.toByteArray(new FileInputStream(tmp));
 
-            addShellStep(String.format(
-                    "base64 --decode << ENDOFFILE > archive.zip && unzip -o archive.zip \n%s\nENDOFFILE",
-                    new String(Base64.encodeBase64Chunked(archive))
-            ));
+            if (SystemUtils.IS_OS_WINDOWS) {
+                if (!(controller instanceof LocalController)) {
+                    // TODO: Make it work for RemoteJenkinsController like in Unix (below)
+                    throw new AssumptionViolatedException("Copying files in Windows is only supported if a LocalController is in use. Test will be skipped.");
+                }
+                addBatchStep("xcopy " + file.getAbsolutePath() + " %cd% /E");
```

Comment:
> I'd say that the implementation for *NIX systems (code below) was done such weird in order to avoid issues with special characters in the path.

@amuniz No, the filename has nothing to do with this. The problem is that with a remote Jenkins controller you cannot simply copy a file from a known path into the workspace, because the source is on the machine running the test code. So the Unix trick is to take the bytestream of the file, Base-64 encode it, and then create a command which _when run on any machine_ will cause that exact binary file to reappear in the current directory (in this case a Jenkins workspace). Whether there is a (portable) Windows equivalent to this trick, I have no idiom.